### DISCUSSION TITLE
ebusd Parameter für arotherm

### DISCUSSION THREAD

**Discussion by cyberthom42** [cyberthom42/vaillant-arotherm-plus#13] [General]
Ich würde im Wiki gern eine Zusammenstellung der sinnvollen ebus-Parameter dokumentieren. Ich stoße immer wieder auf Parameter wo ich noch einmal überlegen muss, wie genau die nun zu interpretieren sind. 
Beispiele sind die Kälte/Wärme Leistungs- und Energie Werte. Ich muss immer wieder überlegen, war das nun die Kälte-Leistung/Energie oder Wärmeleistung/Energie. Des weiteren gibt es einige Parameter, mit denen man die WP sehr gut steuern kann.

**Comment by morphZ** [comment]
Hi @cyberthom42. Erstmal danke, dass du diese Infosammlung hier erstellt hast!

Ich habe mir inzwischen schon ewig viele configs, issues und PRs im [john30/ebusd-configuration](https://github.com/john30/ebusd-configuration) repo durchgelesen, habe aber bisher noch nicht eine konsoledierte high-quality config finden können.

Meine WP besteht aus aroTHERM plus VWL 75/6 A S2, Hydraulikstaton VWZ MEH 97/6, und Regler VRC 720/2.

```
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu00.HW5103.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV2;SW=0514;HW=1104", loaded "vaillant/15.ctlv2.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0607;HW=5103", loaded "vaillant/76.vwzio.HW5103.csv"
```

Ich bin gerne bereit beim Testen, Aufräumen etc. zu unterstützen.

**Reply by cyberthom42** [reply]
Mein Setup ist nahezu identisch. Ja, das wäre super wenn hier jemand unterstützen kann. Ich habe es noch nicht geschafft ein ordentliches Pull über github einzurichten. Ich würde gern hier ein alternatives Reposity anlegen, da die PR scheinbar bei ebusd nicht bearbeitet werden. Das Repository hier ist ja nur für die Arotherm. Ich denke die wichtigste Datei ist die XLS die ich hochgeladen habe. Dort habe ich alle Ergebnisse aus dem Debugging notiert

**Reply by morphZ** [reply]
Ok. Dann schaue ich mir mal das spread sheet an...

**Reply by morphZ** [reply]
Funktionieren bei dir die `YieldThisYearM` und `EnergyConsumptionThisYearM` Nachrichten?


```````
r,,YieldThisYear1,,,,,010C00,month,,D1B,,kWh,,,,IGN:1,,,,,,energy,,,,,,,,,
r,,EnergyConsumptionThisYear1,,,,,020100,month,,D1B,,kWh,,,,IGN:1,,,,,,energy,,,,,,,,,

**Comment by morphZ** [comment]
Frage 1: Ich nutze inzwischen einen Parameter unter B512. Das ist ein Polling den die VWZIO an die HMU00 macht. Hier kommen alle 10 min der Durchfluss und der Heizwasserdruck durch. Ist nützlich um Heizleistung zu berechnen...

Hat das schon einer in der Konfig? (bisher nicht gesehen)
```
# Status message for hydraulics,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
*uw,,,,,,B512,,,,,,,,,,,,,,,,,,,,,,,,,
uw,,StatusHydraulics,,,76,B512,1300,Pressure,m,UCH,10,bar,Systemdruck,Flow,m,UIN,,l/h,Durchflussrate,Unknown,m,UCH,,,,,,,,,
```

**Reply by chrizzzp** [reply]
Geht bei meiner Arotherm Split (mit Hydraulikstation VWL 127/5 IS (`76.vwz00.csv`)  leider nicht.

**Reply by cyberthom42** [reply]
Bei mir sehe ich auch auf B512 Druck und Durchfluss. Allerdings kommt das aller 8 Sekunden und es wird nicht gepollt, sondern die HMU schickt die Nachricht an die 76 ohne vorher gepollt worden zu sein. 
```
2023-12-29 15:31:38.180 [update notice] received unknown MS cmd: 0376b5120613000d5b0300 / 0200ff
2023-12-29 15:31:46.319 [update notice] received unknown MS cmd: 0376b5120613000d5a0300 / 0200ff
2023-12-29 15:31:54.188 [update notice] received unknown MS cmd: 0376b5120613000d5d0300 / 0200ff
```

**Reply by morphZ** [reply]
> Allerdings kommt das aller 8 Sekunden und es wird nicht gepollt, sondern die HMU schickt die Nachricht an die 76 ohne vorher gepollt worden zu sein. 

So isses bei mir auch. Ich war vom polling durch die VWZ wegen der 76 ausgegangen... Stimmt aber nicht.

**Reply by chrizzzp** [reply]
Habe auch noch eine weitere Status-Message bei meiner split gefunden. Der Regler (`ctlv2`) fragt das alle 60s bei der `hmu` ab. Stecken verschiedene Informationen drin, die man aber auch schon in anderen Definitionen findet.

`hcmode.inc:`
```
*r,,,,,,B511,,,,,,,,,,,,,,,,,,,,,,,,,
r,,Status00,,,,,00,S00_SupplyTemp,,temp,,,,WaterPressure,,UCH,10,bar,,S00_CompressorModulation,,percent1,,,,CompressorState,,UCH,0=off;1=Heating Prerun;4=Heating;5=Heating Overrun;110=Defrosting,,,HeatingState,,UCH,8=off;9=Heating,,,field6,,UCH,,,,Defrost,,UCH,0=inactive;32=active,,,S00_CompressorPower,,percent1,,,
```

**Reply by chrizzzp** [reply]
Und noch eine... pusht bei meiner Arotherm Split alle 10s diverse Temperaturen etc. von der Hydraulikstation (`vwz00`) zur `hmu`. Hab's  bei  mir in der `hcmode.inc`:

```
*uw,,,,,,"B512",,,,,,,
uw,,StatusPushVWZ,VWZ data pushed every 10s to HMU,,,,11,,,temp:SupplyTemp;temp:ReturnTemp;temp:CondensorOutletTemp;temp:CondensorInletTemp;press10:WaterPressure;flowrate:SystemFlow;UCH;UCH,,
```
Braucht folgende Template-Definitionen:
```
press10,UCH,10,bar,pressure
flowrate,UIN,,l/h,flowrate
```

**Reply by cyberthom42** [reply]
Cool! Ich würde das mal im Wiki sammeln und dann die CSV anpassen.

**Reply by cyberthom42** [reply]
Diese Push Nachrichten sind nützlich um das eigene Polling zu reduzieren

**Reply by morphZ** [reply]
Die B512 von @chrizzzp funktioniert bei mir nicht. Vermutlich, weil ich eine Monoblock hmu habe...

Die B511 kommt bei mir auch. Tolle Ergänzung, danke!

**Reply by cyberthom42** [reply]
Ich habe auch eine Monoblock (55/6). Bei mir kommen auch B512, aller dings auf 13:
`0376b5120613040d5d0300 / 0200ff`

Ich sehe zudem noch ein Poll für B512:
`1076b512030f0201 / 07030300d1010dff`

Im Log sieht das dann so aus:
```
024-01-13 22:48:10.850 [update notice] received unknown MS cmd: 1076b512030f0201 / 07030300fd010dff
2024-01-13 22:48:12.551 [update notice] received unknown MS cmd: 0376b5120613040d590300 / 0200ff
2024-01-13 22:48:20.893 [update notice] received unknown MS cmd: 1076b512030f0201 / 07030300fd010dff
2024-01-13 22:48:21.018 [update notice] received unknown MS cmd: 0376b5120613040d580300 / 0200ff
2024-01-13 22:48:28.529 [update notice] received unknown MS cmd: 0376b5120613040d5a0300 / 0200ff
2024-01-13 22:48:30.947 [update notice] received unknown MS cmd: 1076b512030f0201 / 07030300f8010dff
2024-01-13 22:48:36.721 [update notice] received unknown MS cmd: 0376b5120613040d5c0300 / 0200ff
2024-01-13 22:48:41.737 [update notice] received unknown MS cmd: 1076b512030f0201 / 07030300ef010dff
2024-01-13 22:48:44.488 [update notice] received unknown MS cmd: 0376b5120613040d5b0300 / 0200ff
2024-01-13 22:48:51.578 [update notice] received unknown MS cmd: 1076b512030f0201 / 07030300eb010dff
```

**Reply by cluk1** [reply]
Vielen Dank für eure Arbeit!!

Ich hätte eine kleine Ergänzung zu B511: CompressorState == 24 ist bei mir Warmwasserbereitung:

```
*r,,,,,,"B511",,,,,,,,,,,,,,,,,,,,,,,,,
r,,Status00,,,,,00,S00_SupplyTemp,,temp,,,,WaterPressure,,UCH,10,bar,,S00_CompressorModulation,,percent1,,,,CompressorState,,UCH,0=off;1=Heating Prerun;4=Heating;5=Heating Overrun;24=Hot Water;110=Defrosting,,,HeatingState,,UCH,8=off;9=Heating,,,field6,,UCH,,,,Defrost,,UCH,0=inactive;32=active,,,S00_CompressorPower,,percent1,,,
```

**Comment by morphZ** [comment]
Frage 2: Weißt du, warum die PRs im Main Repo nicht gemergt werden? Habe das Gefühl, dass dadurch sehr viele Versionen von allem entstehen...

**Reply by cyberthom42** [reply]
Warum kann ich nicht beantworten. Aber das ist einer der Gründe, warum ich selbst eins auf gemacht habe. Es gibt in den vielen Threads schon viele gute Infos. Aber die muss man sich dann selbst in eine Datei gießen :-(
Ich habe dann systematisch alle Bereiche durchgescannt und dann mühselig mit diff versucht herauszufinden was sich ändert. Es werden ja auch z.T. verschiedene Datentypen verwendet.

**Comment by chrizzzp** [comment]
> Ich sehe zudem noch ein Poll für B512:
> `1076b512030f0201 / 07030300d1010dff`

Bei meiner Split-Anlage kann ich das B512-Register `0f0201` manuell von der Hydraulikstation (`vwz00`) pollen. Es kommt die gleiche Antwort wie beim Register `0f0001`, das der Regler (VRC720) in meiner Anlage alle 10 Sekunden von sich aus pollt. Einen Teil der Werte habe ich schon zugeordnet. Haben ggf. alle mit dem Heizstab (immersion heater) zu tun. Die anderen Werte ändern sich gerade nicht bei mir, vielleicht weil ich nie den Heizstab an habe. Falls das mal jemand testen will...

`vwz00.csv`:

`*r,,,,,,B512,,,,,,,,,,,,,,,,,,,,,,,,,
r,,vUnknownB512_0001,,,,,0F0001,,,UCH;UCH;UCH;temp:supplyimmersionheater;press10:WaterPressure;UCH,,`

**Comment by cyberthom42** [comment]
Ich hab das mal alles im Wiki zusammen gefasst

**Comment by tttooommmeeekkk** [comment]
Kennt jemand das Register für die Vorlauf-Soll-Temperatur?



Am Display der Hydraulikstation wird ja sowohl der aktuelle Istwert, als auch der aktuelle Soll-Wert angezeigt. In meiner csv für die `hmu00` finde ich aber keinen `SupplyTempDesired`-Eintrag oder dgl.. Weiß jemand, welchen Wert ich hier nehmen muss?
Akt. Vorlauf ist `SupplyTemp` kann ich finden und der liefert auch plausible Werte.

Oder gibt es den Wert nicht als Register und er wird für die Ausgabe am Display direkt berechnet (analog Wiki hier)?

**Reply by cyberthom42** [reply]
meinst du das hier:
r1,basv,Hc1ActualFlowTempDesired,ActualFlowTempDesired Heizkreis 1,,15,b524,020002000700,,s,IGN:4,,,,,s,EXP,,°C,current flow temperature setpoint of Hc1

**Reply by tttooommmeeekkk** [reply]
Danke. Das liefert offenbar den Wert, der am Systemregler `ctlv` (sensoComfort ohne Kommastelle) für den aktiven Heizkreis angezeigt wird.
Den Wert, der am an der Display meiner Hydraukikeinheit (uniTower) angezeigt wird, ist bei mir jedoch ein anderer. Daher müsste es doch auch für die Außeneinheit `hmu` ein entsprechendes Register geben eben sowas, wie `SupplyTemp` (Vorlauf Ist) eben als `SupplyTempDesired` (Vorlauf Soll) ?!

**Reply by chrizzzp** [reply]
@tttooommmeeekkk 

Vorlaufsolltemperatur der Wärmepumpe kriegst Du z.B. aus `SetMode` (flowtempdesired).

```
uw,,,SetMode,Operation mode,,,b510,00,hcmode,,UCH,0=auto;1=off;2=heat;3=water,,boiler mode,flowtempdesired,,D1C,,°C,temperature,hwctempdesired,,D1C,,°C,temperature,hwcflowtempdesired,,UCH,,°C,temperature,ign,,IGN:1,,,,disablehc,,BI0,,,,disablehwctapping,,BI1,,,,disablehwcload,,BI2,,,,ign_1,,IGN:1,,,,remotecontrolhcpump,,BI0,,,,releasebackup,,BI1,,,,releasecooling,,BI2,,,
```

**Reply by tttooommmeeekkk** [reply]
@chrizzzp Super. Danke, das passt.

Nun stellt sich noch die Frage: Welcher der beiden Werte wird für die Errechnung des Energieintegrals herangezogen?
Und welches ist der jeweilige Referenzwert:
• Systemregler: `hc1flowtemp` <> `hc1actualflowtempdesired`
oder
• Wärmepumpe: `setmode_flowtempdesired`  <> `supplytemp` ? Ich würde auf Letzteres tippen.

**Reply by chrizzzp** [reply]
Ja, letzteres. Es wird ja nicht für jeden Heizkreis ein eigenes Energieintegral berechnet (es gibt ja üblicherweise auch nur einen gemeinsamen Wärmeerzeuger).

**Comment by tttooommmeeekkk** [comment]
Bei einer Temperatur habe ich aktuell noch nein paar Fragezeichen:

Die von der WP gelieferte `**condensationtemp**` kann ich irgendwie nicht richtig verorten. Wo genau im Gerät wird die gemessen? Wenn die WP läuft, steigt sie an (weniger `compressoroutlettemp` aber  größer `condensoroutlettemp` ); also irgendwo hinter Kompressor bei Verflüssiger. Sobald der Kompressor ausgeht, fällt sie rasant (auf in etwa Lufteinlass-Temperatur). Das passt dann irgendwie nicht so in die Ecke des Verflüssigers oder doch?  Da werd ich aktuell nicht schlau draus. Das Funktionsschema (Abbildung in der Anleitung zur Ausseneinheit) liefert mir da auch keinen plausiblen Hinweis (die dort eingezeichneten Temperatursensoren passen irgenwie nicht zu den angezeigten Werten).

Hat da jemand eine Idee oder kann mir die Logik erkkären?

**Reply by chrizzzp** [reply]
Gleiches Problem und gleiche Beobachtungen hier.  Habe eine Split-Anlage und der Verflüssiger ist in der Inneneinheit. Wenn der Kompressor aus ist, zeigen `condensorinlet/condensoroutlet `-Sensoren eher Temperaturen im Bereich des Wasserkreislaufs (also innen), während `condensationtemp` eher die Außentemperatur/Lufteinlass hat. Dazu passt auch, daß `condensorinlet/condensoroutlet`-Temperaturen bei meinem System per ebus von der Inneneinheit (vwz00) kommen, während `condensationtemp` von der Außeneinheit (hmu) kommt. Ich würde letzteren also eher in der Außeneinheit verorten. Dann wäre es aber vermutlich kein Sensor für  `condensationtemp`, sondern hätte eher was mit dem Verdampfer zu tun.

**Reply by tttooommmeeekkk** [reply]
Aber am Verdampfer erreicht man ja nicht die Temperaturen des Wasserkreislaufs. Und die 
`condensationtemp` folgt bei aktiver WP eigentlich schön annähernd der Vorlauf-Ist der AE (hmu).
Schon ein wenig mysteriös (aber vielleicht ist es ja auch nur ein zusammengewürfelter Wert je nach Betriebszustand der Anlage aus untersch. Sensoren).

**Reply by chrizzzp** [reply]
Stimmt, bei mir auch nah am VL-Ist, aber wirklich nur bei aktivem Kompressor. Sonst folgt `condensationtemp` eher der Außentemperatur.

**Comment by tttooommmeeekkk** [comment]
Und noch ein Parameter, den ich nicht verstehe:
`buildingcircuitflow` (Heizwasser-Fließgeschwindigkeit) geliefert von der `hmu` (Außeneinheit)

Eigentlich dachte ich bislang, dass bei Raumaufschaltung = erweitert der Heizkreis geschlossen wird und die Fließgeschwindigkeit bei Erreichen der Raumsolltemp. immer auf 0 sinkt (weil Heizkreis geschlossen).
Bei deaktivierer Raumaufschaltung (also nur heizkurvengesteuert) bleibt der Heizkreis ja geöffnet und die Fließgeschwindigkeit läuft vermindert auch zwischen den Takten weiter.

Nun habe ich heute aber gesehen, dass auch bei aktiver Raumaufschaltung das Heizwasser mal fließt und mal nicht.

Weiß jemand, warum das so ist und welche Faktoren und Schwellwerte ggf. Einfluss darauf haben?

<img width="1224" height="3957" alt="Screenshot_20251114-133946_Home Assistant" src="https://github.com/user-attachments/assets/f4617c24-e5fa-439e-b4f1-a9079e82f90c" />


**Edit**: Habe zwischenzeitlich herausgefunden, dass die Anlage bei Nichtabschaltung der Heizpumpe einen `statuscode` von _Shutdown_ hat und (noch) nicht im _Standby_ ist. Daran wird es dann/wohl liegen. Aber warum fährt sie teilweise mal extrem lange Shutdown-Phasen* von über 1h (gefolgt von neuem Takt ohne _Standby_) und dann wieder kürzere Shutdown-Pahsen** von gerade mal 2min, wo sie es dann auch vor Wiederanlaufen in den Standby-Modus schafft?

.* Ablauf: Compressor active → Overrun (1min)→Compressor blocked (2min) → Shutdown (**1h**) → Prerun (2min) → Compressor active
<img width="1224" height="3298" alt="Screenshot_20251114-193344_Home Assistant" src="https://github.com/user-attachments/assets/5542b1e3-0b51-4721-b8bc-b21883e3626a" />


.* * Ablauf: Compressor Active → Overrun (5min)→~Compressor blocked (0min)~ → ~Shutdown (0min)~ → Standby (xmin)→ Prerun (2min) → Compressor active
<img width="1224" height="3309" alt="Screenshot_20251114-193431_Home Assistant" src="https://github.com/user-attachments/assets/fef6ca28-e524-4eb6-bce6-2b864ac9e81b" />


Also irgendwie liegt es dann an dem (nicht) abgebauten EnergieIntegral. Vorlauf-Soll der `SetMode` bleibt manchmal bestehen (EI kann daher abgebaut werden, damit KEIN Standby, sondern Shutdown);

manchmal geht der Vorlauf-Soll auf 0 (EI kann NICHT abgebaut werden, daher Standby).

Ich verstehe das System (SetMode) hier nicht bzw. erkenne kein System.
